src/apps/api/views/profiles.py

```diff
@@ -6,12 +6,37 @@
 from django.http import HttpResponse
 from rest_framework.generics import GenericAPIView, RetrieveAPIView
 from rest_framework import permissions
+from rest_framework.viewsets import ModelViewSet
+from rest_framework.response import Response
+from django.urls import reverse
 
-from api.serializers.profiles import MyProfileSerializer
+from api.permissions import IsUserAdminOrIsSelf
+from api.serializers.profiles import MyProfileSerializer, UserSerializer
 
 User = get_user_model()
 
 
+class UserViewSet(ModelViewSet):
+    queryset = User.objects.all()
+
+    def get_serializer_class(self):
+        return UserSerializer
+
+    def get_permissions(self):
+        if self.action in ['update', 'partial_update']:
```

Comment:
This guy right here is what leaves the `destroy` vulnerable to being hit by any authenticated user.